src/models/requestVariableCacheValue.ts

```diff
@@ -0,0 +1,11 @@
+"use strict";
+
+import { HttpRequest } from "./httpRequest";
+import { HttpResponse } from "./httpResponse";
+
+export class RequestVariableCacheValue {
+    public constructor(
+        public request: HttpRequest,
+        public response?: HttpResponse) {
```

Comment:
Why add a `?` here, when response doesn't meet?

Reply:
The idea at first was; cache request as it is run, then update cache with response when it is returned. But I dropped the idea since the use of the variable sort of depends on the response completing. Forgot to revert the `?`, will do now.